data_model/src/visit.rs

```diff
@@ -0,0 +1,752 @@
+//! Visitor that visits every node in Iroha syntax tree
+#![allow(missing_docs, clippy::missing_errors_doc)]
+
+#[cfg(not(feature = "std"))]
+use alloc::format;
+
+use crate::{evaluate::ExpressionEvaluator, prelude::*, NumericValue};
+
+macro_rules! delegate {
+    ( $($visitor:ident($operation:ty)),+ $(,)? ) => { $(
+        fn $visitor(&mut self, authority: &AccountId, operation: $operation) {
+            $visitor(self, authority, operation);
+        } )+
+    }
+}
+
+macro_rules! deny_unsupported_instruction {
+    ($isi_type:ty) => {
+        unimplemented!() // concat!("Unsupported `", stringify!($isi_type), "` instruction").to_owned())
```

Comment:
I've resorted to using `panic` atm. We might want to change this in

Reply:
we don't really need a test for this, it's impossible for unsupported instruction to execute

Reply:
Ok, so we guarantee that in our code. And if somehow we reach this line then we panic, because it's a bug and should never be reached?

Then I wonder if `unimplemented!()` is a correct macro here... I'd also consider `unreachable!()`. Because we don't really plan to impelement it one day. We just assume it will never happen in our code path

Reply:
Feel free to close this discussion

Reply:
Oh, I see, it's outdated. I close this